purchases-sample/src/main/java/com/revenuecat/sample/OverviewFragment.kt

```diff
@@ -0,0 +1,91 @@
+package com.revenuecat.sample
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.doOnPreDraw
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.FragmentNavigatorExtras
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.google.android.material.transition.MaterialElevationScale
+import com.revenuecat.purchases.EntitlementInfo
+import com.revenuecat.purchases.Offering
+import com.revenuecat.purchases.Offerings
+import com.revenuecat.purchases.Purchases
+import com.revenuecat.purchases.getOfferingsWith
+import com.revenuecat.purchases.getPurchaserInfoWith
+import com.revenuecat.sample.databinding.FragmentOverviewBinding
+
+class OverviewFragment : Fragment(), OfferingCardAdapter.OfferingCardAdapterListener {
+
+    lateinit var binding: FragmentOverviewBinding
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        binding = FragmentOverviewBinding.inflate(inflater)
+
+        binding.purchaserInfoCard.setOnClickListener {
+            with(binding.purchaserInfoDetailsContainer) {
+                visibility = if (visibility == View.GONE) View.VISIBLE else View.GONE
```

Comment:
Just curious, why is the visibility set to GONE in the xml and then always set to VISIBLE here?

Reply:
we want it gone when the user launches the app. but on every click it will toggle visibility, expanding or collapsing the JSON object section